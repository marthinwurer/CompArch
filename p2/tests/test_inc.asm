;
; tests INC
;
; uses instructions BPT, HALT, MOV
;
; uses addressing modes 0, 1, 2, 6
;
	.globl	start
;
one:	.word	1
;
. = . + 50
ptr:	.word	alpha
. = . + 10
alpha:	.word	42
. = . + 10
;
start:	inc	r1
	bpt
;
	inc	one
	mov	one,r2
	bpt
;
	inc	one
	inc	one
	mov	one,r2
	bpt
;
	mov	ptr,r3
	inc	(r3)
	mov	alpha,r4
	bpt
;
	inc	(r3)+
	mov	alpha,r5
	bpt
;
	halt
;
	.end	start
