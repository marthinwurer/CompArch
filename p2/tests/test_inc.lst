      1                                 ;
      2                                 ; tests INC
      3                                 ;
      4                                 ; uses instructions BPT, HALT, MOV
      5                                 ;
      6                                 ; uses addressing modes 0, 1, 2, 6
      7                                 ;
      8 000000                                   .globl  start
      9                                 ;
     10 000000  000001                  one:     .word   1
     11                                 ;
     12 000052                          . = . + 50
     13 000052  000064'                 ptr:     .word   alpha
     14 000064                          . = . + 10
     15 000064  000042                  alpha:   .word   42
     16 000076                          . = . + 10
     17                                 ;
     18 000076  005201                  start:   inc     r1
     19 000100  000003                           bpt
     20                                 ;
     21 000102  005267  177672                   inc     one
     22 000106  016702  177666                   mov     one,r2
     23 000112  000003                           bpt
     24                                 ;
     25 000114  005267  177660                   inc     one
     26 000120  005267  177654                   inc     one
     27 000124  016702  177650                   mov     one,r2
     28 000130  000003                           bpt
     29                                 ;
     30 000132  016703  177714                   mov     ptr,r3
     31 000136  005213                           inc     (r3)
     32 000140  016704  177720                   mov     alpha,r4
     33 000144  000003                           bpt
     34                                 ;
     35 000146  005223                           inc     (r3)+
     36 000150  016705  177710                   mov     alpha,r5
     37 000154  000003                           bpt
     38                                 ;
     39 000156  000000                           halt
     40                                 ;
     41 000160  000076'                          .end    start


alpha    000064 R  
one      000000 R  
pc     = 000007 %  
ptr      000052 R  
r0     = 000000 %  
r1     = 000001 %  
r2     = 000002 %  
r3     = 000003 %  
r4     = 000004 %  
r5     = 000005 %  
r6     = 000006 %  
r7     = 000007 %  
sp     = 000006 %  
start    000076 RG 
