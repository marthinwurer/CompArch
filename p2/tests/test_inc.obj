000 2 01 00
02a 2 34 00
034 2 22 00
03e 2 81 0a
040 2 03 00
042 4 b7 0a ba ff
046 4 c2 1d b6 ff
04a 2 03 00
04c 4 b7 0a b0 ff
050 4 b7 0a ac ff
054 4 c2 1d a8 ff
058 2 03 00
05a 4 c3 1d cc ff
05e 2 8b 0a
060 4 c4 1d d0 ff
064 2 03 00
066 2 93 0a
068 4 c5 1d c8 ff
06c 2 03 00
06e 2 00 00
03e
