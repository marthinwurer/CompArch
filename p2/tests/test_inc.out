CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 76
000076:  PS[00]  005201  INC R1
000100:  PS[00]  000003  BPT
  R0[000000]  R1[000001]  R2[000000]  R3[000000]
  R4[000000]  R5[000000]  R6[000000]  R7[000102]
000102:  PS[00]  005267 177672  INC 177672(R7)
000106:  PS[00]  016702 177666  MOV 177666(R7),R2
000112:  PS[00]  000003  BPT
  R0[000000]  R1[000001]  R2[000002]  R3[000000]
  R4[000000]  R5[000000]  R6[000000]  R7[000114]
000114:  PS[00]  005267 177660  INC 177660(R7)
000120:  PS[00]  005267 177654  INC 177654(R7)
000124:  PS[00]  016702 177650  MOV 177650(R7),R2
000130:  PS[00]  000003  BPT
  R0[000000]  R1[000001]  R2[000004]  R3[000000]
  R4[000000]  R5[000000]  R6[000000]  R7[000132]
000132:  PS[00]  016703 177714  MOV 177714(R7),R3
000136:  PS[00]  005213  INC (R3)
000140:  PS[00]  016704 177720  MOV 177720(R7),R4
000144:  PS[00]  000003  BPT
  R0[000000]  R1[000001]  R2[000004]  R3[000064]
  R4[000043]  R5[000000]  R6[000000]  R7[000146]
000146:  PS[00]  005223  INC (R3)+
000150:  PS[00]  016705 177710  MOV 177710(R7),R5
000154:  PS[00]  000003  BPT
  R0[000000]  R1[000001]  R2[000004]  R3[000066]
  R4[000043]  R5[000044]  R6[000000]  R7[000156]
000156:  PS[00]  000000  HALT

Machine Halted - HALT instruction

Simulated time 131 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
